### DISCUSSION TITLE
How to customize cursor for typing in jquery terminal

### DISCUSSION THREAD

**Discussion by Thecode764** [jcubic/jquery.terminal#948] [General]
Hello in the jquery terminal you can use `prompt:` for set prompt for terminal how to customize the cursor for typing in jquery terminal
![Screenshot_20240517_093133](https://github.com/jcubic/jquery.terminal/assets/141347375/f1ea4fab-f0e2-42a3-bcbf-b1818317386a)
the cursor for typing

**Comment by jcubic** [comment]
You can use CSS variable:

```
--animation
```

To change the cursor animation:

There are 4 builtin animations:

* `--animation: terminal-blink;` - default
* `--animation: terminal-none;` - no animation
* `--animation: terminal-bar;` - vertical bar like in Input box
* `--animation: terminal-underline;` - like in Windows cmd.exe

If you want to create your own animation you need to look a the source code.

To change the color of the cursor without affecting the rest of the terminal, you can use this:

```
.cmd-cursor {
    --color: white;
}
```

Here is an example of [changing colors on different parts of the terminal](https://codepen.io/jcubic/pen/oNXqJWY).
